### PR TITLE
[windows] non-blocking TransmitFile

### PR DESCRIPTION
**Pull request by cirospaciari**
### What does this PR do?

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

Existing Tests

### CODE REVIEW

File: src/bun.js/api/server.zig [hunk 1]
```diff
@@ -86,6 +86,7 @@ const SendfileContext = struct {
     has_listener: bool = false,
     has_set_on_writable: bool = false,
     auto_close: bool = false,
+    transmitFileContext: if (Environment.isWindows) ?bun.windows.TransmitFileContext else u0 = if (Environment.isWindows) null else 0,
```
Comment (jdalton) [inline src/bun.js/api/server.zig:89]: This chained ternary-ish-conditional is hard to parse mentally. It looks like the `else` side of `if (Environment.isWindows)` so the `false` side also has a check for `if (Environment.isWindows)`.

File: src/bun.js/api/server.zig [hunk 2]
```diff
@@ -1886,13 +1887,27 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
...
             } else if (Environment.isWindows) {
-                const win = std.os.windows;
-                const uv = bun.windows.libuv;
-                const socket = bun.socketcast(this.sendfile.socket_fd);
-                const file_handle = uv.uv_get_osfhandle(bun.uvfdcast(this.sendfile.fd));
-                this.sendfile.offset += this.sendfile.remain;
-                this.sendfile.remain = 0;
-                return win.ws2_32.TransmitFile(socket, file_handle, 0, 0, null, null, 0) == 1;
+                if (this.sendfile.transmitFileContext == null) {
+                    this.sendfile.transmitFileContext = bun.windows.TransmitFileContext.init();
```
Comment (jdalton) [inline src/bun.js/api/server.zig:1891]: We have a `bun.windows` and ` bun.win32` struct which is confusing.

**Review by Jarred-Sumner — CHANGES_REQUESTED** [review]
fetch tests are failing

**Comment by Jarred-Sumner** [comment]
@cirospaciari which tests are now passing because of this PR?
